src/VisualStudio/Core/Def/Implementation/Venus/ContainedDocument.cs

```diff
@@ -136,6 +139,13 @@ private HostType GetHostType()
             var projectionBuffer = _containedLanguage.DataBuffer as IProjectionBuffer;
             if (projectionBuffer != null)
             {
+                // Projection buffers can have source buffers of multiple supported types. 
+                // First check to see if the subject buffer is the TypeScript content type.
+                if (_containedLanguage.SubjectBuffer.ContentType.IsOfType(TypeScript))
```

Comment:
Thanks - it makes sense that the HpostType refers to the document you’re contained within.  
For HTML docs with JS script blocks (using the TSLS) the projection buffer’s source buffer ContentTypes are: "HTMLX", "TypeScript"

Since the code currently only checks for “HTML” and “Razor” it is throwing ExceptionUtilities.Unreachable. 

Instead, I could also check if the Content.TypeName is “HTMLX”:

if (projectionBuffer.SourceBuffers.Any(b => b.ContentType.IsOfType(HTML) || string.Equals(b.ContentType.TypeName, HTMLX, StringComparison.OrdinalIgnoreCase)))
{
   return HostType.HTML;
}

Does this sound ok?

Note:  I can’t use IsOfType(HTMLX) as RazorCSharp  derives from HTMLX so it would be marked as HTML.  I also can't check for IsOfType(TypeScript) as Razor docs could also contain TypeScript subject buffer content types.